Formula/dotnet.rb

```diff
@@ -0,0 +1,33 @@
+class Dotnet < Formula
+  desc ".Net Core"
+  homepage "https://dotnet.microsoft.com/"
+  url "https://github.com/dotnet/source-build.git",
+      tag:      "v3.1.108-SDK",
+      revision: "05a0c8f6b13fcbec4b441c977b639b75e8a74f67"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "curl" => :build
+  depends_on "pkg-config" => :build
+  depends_on "icu4c"
+  depends_on "openssl"
+
+  def install
+    # Arguments needed to not artificially time-limit downloads from Azure.
+    # See the following GitHub issue comment for details:
+    # https://github.com/dotnet/source-build/issues/1596#issuecomment-670995776
+    reference_timeout = "/p:DownloadSourceBuildReferencePackagesTimeoutSeconds=N/A"
+    artifact_timeout = "/p:DownloadSourceBuiltArtifactsTimeoutSeconds=N/A"
+    system "./build.sh", reference_timeout, artifact_timeout
+
+    lib.mkpath
+    tarball = Dir["artifacts/*/Release/dotnet-sdk-#{version}-*.tar.gz"].first
+    system "tar", "-xzf", tarball, "--directory", lib
+
+    (bin/"dotnet").write_env_script lib/"dotnet", DOTNET_ROOT: ENV.fetch("DOTNET_ROOT", lib)
+  end
+
+  test do
+    system bin/"dotnet", "--info"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks, I've added a much more elaborate test now that writes a project file and a source C# file to disk, compiles it, runs the compiled binary and then checks the output.